src/phalanx/cli.py

```diff
@@ -133,6 +135,44 @@ def help(ctx: click.Context, topic: str | None, subtopic: str | None) -> None:
     raise click.UsageError(msg, ctx)
 
 
+@main.group()
+def application() -> None:
+    """Commands for Phalanx application configuration."""
+
+
+@application.command("create")
+@click.argument("name")
+@click.argument("starter")
```

Comment:
Could the CLI document what starters are available?

I also wonder if the starter should be an option with `web-service` as the default? The nice thing about it being an option is that it could be a [`choices`](https://click.palletsprojects.com/en/8.1.x/options/#choice-options) type. To implement it we'd need to have a module-scoped service that looks up the directory names in the `starters` directory, so it isn't the cleanest, but it might be a better user experience.

Reply:
So, the reason why this is tricky is that the location of the configuration can be specified with a command-line flag, so it's not known at the module level, which means I can't determine the available starters at the module level. I'm not sure there's any way around that while still supporting the `-c` flag. That's why I got stuck on runtime errors.

That said, I agree that it's a real downside to the interface, so I think what I'm going to do instead is just hard-code the available starters in an enum and then add a test so that tests fail if someone adds a new starter and forgets to add it to the enum. It's not as good as dynamic discovery, but it gets us better help output.